dashboard/modules/reporter/reporter_head.py

```diff
@@ -154,6 +155,49 @@ async def get_cluster_status(self, req):
             cluster_status=formatted_status if formatted_status else None,
         )
 
+    @routes.get("/worker/traceback")
+    async def get_traceback(self, req) -> aiohttp.web.Response:
+        if "ip" in req.query:
+            reporter_stub = self._stubs[req.query["ip"]]
+        else:
+            reporter_stub = list(self._stubs.values())[0]
+        pid = int(req.query["pid"])
+        reply = await reporter_stub.GetTraceback(
+            reporter_pb2.GetTracebackRequest(pid=pid)
+        )
+        logger.info(reply)
+        if reply.success:
+            return aiohttp.web.Response(text=reply.output)
+        else:
+            return aiohttp.web.HTTPInternalServerError(text=reply.output)
+
+    @routes.get("/worker/cpu_profile")
+    async def cpu_profile(self, req) -> aiohttp.web.Response:
+        if "ip" in req.query:
+            reporter_stub = self._stubs[req.query["ip"]]
+        else:
+            reporter_stub = list(self._stubs.values())[0]
+        pid = int(req.query["pid"])
+        duration = int(req.query.get("duration", 5))
+        if duration > 60:
+            raise ValueError(f"The max duration allowed is 60: {duration}.")
+        format = req.query.get("format", "flamegraph")
+        reply = await reporter_stub.CpuProfiling(
+            reporter_pb2.CpuProfilingRequest(pid=pid, duration=duration, format=format)
+        )
+        logger.info(reply)
+        if reply.success:
+            return aiohttp.web.Response(
+                body=reply.output,
```

Comment:
Do you know what's the max output reply size for aiohttp? Maybe we can log warning if the size is too big?

Reply:
Added some logging improvements for these two handlers.